src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -0,0 +1,153 @@
+
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    courtCourtTypes: CourtType[] = null): Promise<void> {
+    if (!courtCourtTypes) {
+
+      const slug: string = req.params.slug as string;
+      await req.scope.cradle.api.getCourtCourtTypes(slug)
+        .then((value: CourtType[]) => courtCourtTypes = value)
+        .catch(() => error += this.getCourtTypesErrorMsg);
+
+    }
+
+    let allCourtTypes: CourtType[] = [];
+
+    await req.scope.cradle.api.getCourtTypes()
+      .then((value: CourtType[]) => allCourtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+
+    const pageData: CourtTypePageData = {
+      errorMsg: error,
+      updated: updated,
+      items: this.mapCourtTypeToCourtTypeItem(allCourtTypes, courtCourtTypes)
+
+    };
+
+    res.render('courts/tabs/typesContent', pageData);
+  }
+
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let courtCourtTypes: CourtType[] = null;
+
+    if(!CSRF.verify(req.body._csrf)) {
+      return this.get(req, res, false, this.updateErrorMsg, courtCourtTypes);
+    }
+
+    if (req.body.types) {
+
+      courtCourtTypes = this.mapBodyToCourtType(req.body);
+
+      if(courtCourtTypes.find( c => (c.name ==="Magistrates' Court" && this.CheckCodeIsNullOrNan(c.code) )
+      || (c.name==='County Court' && this.CheckCodeIsNullOrNan(c.code))
+      || (c.name === 'Crown Court' && this.CheckCodeIsNullOrNan(c.code)) )){
+
+        return this.get(req, res, false, this.emptyCourtCodeErrorMsg, courtCourtTypes);
+      }
+
+      else
+      {
+        await req.scope.cradle.api.updateCourtCourtTypes(req.params.slug, courtCourtTypes)
+          .then((value: CourtType[]) => this.get(req, res, true, '', value))
+          .catch(() => this.get(req, res, false, this.updateErrorMsg, courtCourtTypes));
+      }
+    }
+    else
+    {
+      return this.get(req, res, false, this.emptyCourtTypesErrorMsg, courtCourtTypes);
+    }
+
+  }
+
+  private mapCourtTypeToCourtTypeItem(allCourtTypes: CourtType[], courtCourtTypes: CourtType[]): CourtTypeItem[] {
+
+    if( courtCourtTypes && allCourtTypes) {
+      const courtTypeItems = allCourtTypes.map((ott: CourtType) => (
+        {
+          value: ott.id + '_&_' + ott.name,
+          text: ott.name,
+          checked: this.isChecked(ott, courtCourtTypes),
+          code: this.getCode(ott.id, courtCourtTypes)
+        }));
+
+      return courtTypeItems;
+    }
+    else
+      return [];
+  }
+
+
+  private mapBodyToCourtType(body: any): CourtType[] {
+
+    const courtTypes: string[] = Array.isArray(body.types) ? body.types : [body.types];
+
+    const courtTypeItems = courtTypes.map((ct) => (
+      {
+        id: parseInt(ct.split('_&_',2)[0]),
+        name: ct.split('_&_',2)[1],
+        code:this.setCode(ct.split('_&_',2)[1], body.magistratesCourtCode, body.countyCourtCode, body.crownCourtCode),
```

Comment:
Do we need to do the concatenation of id and name and then this splitting of the string?

Could the 'value' (line 90) just be the id? Do we need the name to be posted to FaCT API given that the search_courtCourtType table just needs the id?

Reply:
removed split string function